⬆ UPDATE: myst-nb 0.11

choldgraf opened the pull request:
This is an update of myst-nb that brings in the new myst parser syntax + config.

# Changed things

- myst-nb to v0.11
- documented "substitutions"
- updated `config.py` logic to use new myst configuration pattern + a test
- updated docs to use new myst configuration pattern

# ToDo

- [x] Tests should pass
- [x] Decide if we need to add any extra tests
- [x] Update documentation for any new syntax + config

choldgraf commented:
OK I _think_ this is ready to go, but I might be missing something. @chrisjsewell could you take a look and make sure I'm not missing any important functionality from myst-nb/myst-parser that I forgot to port over here or document?

@mmcky and @AakashGfude wanna take a look at the docs updates / implementation and let me know if it looks reasonable to you?

mmcky commented on docs/content/content-blocks.md:
````diff
@@ -79,22 +130,12 @@ My content
-parse:
-  myst_extended_syntax: true
-```
+However, if you're using an interface that does not support {term}`MyST Markdown`, it will render as a "raw" code block.
````
@choldgraf does jupytext treat the as `code` blocks? They remain contained in the markdown right and will be shown literally? Perhaps the use of `code block` has confused me.

mmcky replied:
I think I know what you mean here -- perhaps "raw literal block"?

mmcky commented on docs/content/content-blocks.md:
```diff
@@ -359,14 +384,16 @@ own titles and stylings. For example:
 ^^^
-:::{admonition,dropdown,tip} Click here!
+:::{admonition} Click here!
+:class: tip, dropdown
```
@choldgraf I think the use of `class` here is great and makes the recommended syntax a lot clearer.

mmcky commented on docs/content/myst.md:
```diff
@@ -159,7 +160,9 @@ For a list of directives that are available to you, there are three places to ch
 3. This documentation has several additional directives that are specific to Jupyter Book.
 
-In some unusual cases, MyST-Parser may be incompatible with a certain role or directive.
+:::{admonition} What if it exists in rST but not MyST?
```
@choldgraf we could also just use `:::{tip}` in this case right?

mmcky replied:
I have been using

````
```{tip}
````

so perhaps `:::` is different.

choldgraf replied:
Yes with one exception, the `{tip}`, `{note}` etc admonitions don't let you specify a custom title (in those cases, the title is "tip", "note", etc). So this is a way to add your own title.

And ` ```{tip} ` should work the exact same way as `:::{tip}` now

mmcky commented:
thanks @choldgraf for putting this together. Just added some quick comments / clarifications re: `docs`

chrisjsewell reviewed (changes requested):
I can still find references to `myst_extended_syntax` in the docs:

```
docs/advanced/sphinx.md:
  277  parse:
  278:   myst_extended_syntax: true
  279  ```

docs/content/content-blocks.md:
  414  parse:
  415:   myst_extended_syntax: true
  416  ```

docs/content/figures.md:
   52  parse:
   53:   myst_extended_syntax: true
   54  ```

  137  parse:
  138:   myst_extended_syntax: true
  139  ```

docs/content/math.md:
  71  parse:
  72:   myst_extended_syntax: true
  73  ```

docs/content/myst.md:
  250  parse:
  251:   myst_extended_syntax: true
  252  ```

docs/file-types/markdown.md:
  111  parse:
  112:   myst_extended_syntax: true
  113  ```
```

The `html_admonition` syntax extension is also not documented

choldgraf commented:
thanks @mmcky for the comments and questions - I think I resolved them all

thanks @chrisjsewell for catching all of those missing docs, for some reason the trailing `:` was causing my vscode "find" not to find those instances 😬. I think that I've resolved this now, though you should check that I have correctly changed this configuration value in the JSON schema.

choldgraf commented:
I think this is ready to go? Will merge soonishfully unless somebody suggests I do not do this, so that we don't get merge conflicts and so folks have some time to try this on master

mmcky commented:
thanks @choldgraf

chrisjsewell commented:
Yeh it looks good to me thanks @choldgraf, but I would suggest that you should release as soon as this is merged, since otherwise the documentation will be incorrect for the current release. 

I would even suggest you have prepared and reviewed the change log section now, before this is merged, so we can simultaneously merge both and release

chrisjsewell commented:
If you want people to test it first, then I would consider making and alpha release, directly from this branch. I can do this if you are not sure how

choldgraf commented:
@chrisjsewell that's a good point, I will add the changelog to this PR over the weekend and we can fire off a release

choldgraf commented:
OK I've got the changelog updated!

the only thing I was considering doing was waiting on upgrading `sphinxcontrib-bibtex` (https://github.com/executablebooks/jupyter-book/pull/1190) before cutting a new release, because that'll be another deprecation and version bump needed. I think that we could pin it to 2.1 and then just get that PR merged quickly, then rebase here and release, and then make a follow-up PR + minor release for the extra features in v2.2.0 since that won't be a deprecation

choldgraf commented:
OK I've merged #1190, rebased here, and updated the changelog.

Are we 👍 on merging this and then cutting a release? @chrisjsewell I'll await you "un-blocking" the merge per your review before doing this.

chrisjsewell commented on CHANGELOG.md:
````diff
@@ -1,5 +1,96 @@
+    myst_enable_extensions:
+      - <list-of-extensions>
+  ```
+  See [](content-blocks:myst-extensions) for more information.
````
I mentioned to @mmcky last release (https://github.com/executablebooks/jupyter-book/pull/1129#discussion_r538992853), you should perhaps try to avoid reference links in the changelog, since people are likely to also read this document directly  on Github (i.e. from https://github.com/executablebooks/jupyter-book/blob/master/CHANGELOG.md), where such links will be unusable.
e.g. instead something like this might be better:

`[the MyST syntax extension section here](docs/content/content-blocks.md)`

Note, there is actually the option in myst-parser to activate header anchors: https://myst-parser.readthedocs.io/en/latest/using/syntax-optional.html#auto-generated-header-anchors, but this is not "turned on" (or documented) in jupyter book

chrisjsewell commented on CHANGELOG.md:
```diff
@@ -1,5 +1,96 @@
+: The MyST-NB and MyST-parser have both been upgraded.
+  This comes with support for new syntax and a new configuration mechanism (see below for some examples).
+
+  See [the myst-parser changelog](https://myst-parser.readthedocs.io/en/latest/develop/_changelog.html#id4) for more information about the syntax additions.
```
```suggestion
  (See also [the myst-parser changelog](https://myst-parser.readthedocs.io/en/latest/develop/_changelog.html#id4) for more information about the syntax additions.)
```

choldgraf commented:
@chrisjsewell the only reason I didn't merge this straight away is because I'm waiting for you to un-block this PR, you still have "changes requested" and I'd like to respect this until you explicitly say you're happy with the PR.

chrisjsewell commented on CHANGELOG.md:
```diff
@@ -1,5 +1,96 @@
+
+### New
+
+**MyST Parser version 0.13.x**
```
You don't really need to specifically make these definition headers bold, they already "wrapped" in `<dt>` tags and so the theme will take care of them

chrisjsewell commented:
> I think that you've implicitly given your OK to merge, so I'm gonna go ahead and do so, and cut a release so that we don't have outdated docs.

Oh man I was literally half way through the review 🤦

chrisjsewell commented:
> Apologies that the docs are now out of date, I should have gotten the confirmation that you're happy to merge here first before merging the other one.

Yeh its not the end of the world, I wouldn't have mentioned it if was not for the discussion issue. It was just to point out that this can definitely be a problem for users

choldgraf commented:
@chrisjsewell I'll make a PR on master for your suggestions here, sorry for the interference

chrisjsewell commented:
> I'll make a PR on master for your suggestions here

I'm already doing it

choldgraf commented:
@chrisjsewell sounds good 👍
